### PR TITLE
WIP : Refactor dispatch to begin RPC and REST separation.

### PR DESCRIPTION
**Pull request by meriouma**
Also moves toward java 8 api

### CODE REVIEW

File: gwtp-core/pom.xml [hunk 1]
```diff
@@ -27,7 +27,7 @@
         <module>gwtp-dispatch-rpc-server-spring</module>
         <module>gwtp-dispatch-rpc-test</module>
         <module>gwtp-tester</module>
-        <module>gwtp-processors</module>
+        <!--<module>gwtp-processors</module>-->
```
Comment (olafleur) [inline gwtp-core/pom.xml]: remove it instead of commenting it?

**Comment by olafleur** [comment]
LGTM

File: gwtp-core/gwtp-dispatch-common-client/src/main/java/com/gwtplatform/dispatch/client/DispatchCall.java [hunk 2]
```diff
@@ -35,11 +34,11 @@
  * @param <A> The type of the {@link TypedAction} wrapped by this {@link DispatchCall}.
  * @param <R> The type of the result of the wrapped {@link TypedAction}.
  */
-public abstract class DispatchCall<A extends TypedAction<R>, R> {
+public abstract class DispatchCall<A extends TypedAction<R>, R, C> {
```
Comment (Chris-V) [inline gwtp-core/gwtp-dispatch-common-client/src/main/java/com/gwtplatform/dispatch/client/DispatchCall.java]: You should update the javadoc with the C param

File: gwtp-core/gwtp-dispatch-common-client/src/main/java/com/gwtplatform/dispatch/client/DispatchCall.java [hunk 3]
```diff
@@ -130,32 +129,17 @@ protected String getSecurityCookie() {
...
      * @param caught the caught {@link Throwable}.
      */
-    protected void onExecuteFailure(Throwable caught) {
-        if (exceptionHandler != null && exceptionHandler.onFailure(caught) == Status.STOP) {
-            return;
-        }
-
-        callback.onFailure(caught);
+    protected boolean shouldHandleFailure(Throwable caught) {
+        return  !(exceptionHandler != null && exceptionHandler.onFailure(caught) == Status.STOP);
```
Comment (Chris-V) [inline gwtp-core/gwtp-dispatch-common-client/src/main/java/com/gwtplatform/dispatch/client/DispatchCall.java]: 2 spaces after return

Comment (Chris-V) [inline gwtp-core/gwtp-dispatch-common-client/src/main/java/com/gwtplatform/dispatch/client/DispatchCall.java]: ExceptionHandler should never be null. The default binding is DefaultExceptionHandler (it always return CONTINUE)

File: gwtp-core/gwtp-dispatch-common-client/src/main/java/com/gwtplatform/dispatch/client/interceptor/AbstractInterceptor.java [hunk 4]
```diff
@@ -22,7 +22,7 @@
  * @param <A> action class.
  * @param <R> result class.
  */
-public abstract class AbstractInterceptor<A, R> implements Interceptor<A, R> {
+public abstract class AbstractInterceptor<A, R, C> implements Interceptor<A, R, C> {
```
Comment (Chris-V) [inline gwtp-core/gwtp-dispatch-common-client/src/main/java/com/gwtplatform/dispatch/client/interceptor/AbstractInterceptor.java]: Update the javadoc with C

**Comment by Chris-V** [comment]
LGTM, can't wait to be free from the constraints introduced here.

**Pull request merged**